server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -270,6 +313,61 @@ public ClusterMetadataManifest writeIncrementalMetadata(
         return manifest;
     }
 
+    /**
+     * Uploads provided ClusterState's global Metadata to remote store in parallel.
+     * The call is blocking so the method waits for upload to finish and then return.
+     *
+     * @param clusterState current ClusterState
+     * @return String file name where globalMetadata file is stored.
+     */
+    private String writeGlobalMetadata(ClusterState clusterState) throws IOException {
+
+        AtomicReference<String> result = new AtomicReference<String>();
+        final BlobContainer globalMetadataContainer = globalMetadataContainer(
+            clusterState.getClusterName().value(),
+            clusterState.metadata().clusterUUID()
+        );
+        final String globalMetadataFilename = globalMetadataFileName(clusterState.metadata());
+
+        // latch to wait until upload is not finished
+        CountDownLatch latch = new CountDownLatch(1);
+
+        LatchedActionListener completionListener = new LatchedActionListener<>(ActionListener.wrap(resp -> {
+            logger.trace(String.format(Locale.ROOT, "GlobalMetadata uploaded successfully."));
+            result.set(globalMetadataContainer.path().buildAsString() + globalMetadataFilename);
+        }, ex -> {
+            logger.error(() -> new ParameterizedMessage("Exception during transfer of GlobalMetadata to Remote {}", ex.getMessage()), ex);
+            throw new GlobalMetadataTransferException(ex.getMessage(), ex);
+        }), latch);
+
+        GLOBAL_METADATA_FORMAT.writeAsync(
+            clusterState.metadata(),
+            globalMetadataContainer,
+            globalMetadataFilename,
+            blobStoreRepository.getCompressor(),
+            completionListener,
+            FORMAT_PARAMS
+        );
+
+        try {
+            if (latch.await(GLOBAL_METADATA_UPLOAD_WAIT_MILLIS, TimeUnit.MILLISECONDS) == false) {
+                GlobalMetadataTransferException ex = new GlobalMetadataTransferException(
```

Comment:
TODO for emitting metrics for timeouts

Reply:
Created and added task in Meta issue: #10687